src/runtime/type.ts

```diff
@@ -369,6 +369,12 @@ export class EntityType extends Type {
   crdtInstanceConstructor() {
     return this.entitySchema.crdtConstructor();
   }
+
+  handleConstructor<T>() {
+    // Currently using SingletonHandle as the implementation for Entity handles.
+    // TODO: Make an EntityHandle class that uses the proper Entity CRDT.
+    return SingletonHandle;
```

Comment:
I think this will cause problems: the crdtConstructor just above is an EntityCRDT. So you now have a singleton handle connected to a storageproxy with an EntityCRDT, but singleton and entity have different ops.
A test exercising this path would help I think (create an entity type, then create a handle and storage proxy for it and confirm they work).

Reply:
Could you help me write that test? I'm not really sure how to connect up a handle and storage proxy properly. (I could use singletonHandleForTest but that seems like it might bypass the functionality you actually want to test)

Reply:
I think something like this:
https://github.com/PolymerLabs/arcs/blob/e6f86d77ca7b94e1f4d5b161ae22fe0d7bb99f37/src/runtime/storageNG/tests/handle-test.ts#L45
but instead of singletontype use entitytype:

const handle = handleNGFor(
      'me',
      new StorageProxy(
          'id',
          new MockStore // or maybe pass a pec in here?
          EntityType.make([], {})),
      IdGenerator.newSession(),
      fakeParticle,
      true,
      true);
then try set/get the entity